src/env/set.c

```diff
@@ -0,0 +1,42 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*   tosh - 21 Shell                                      ::::::::            */
+/*                                                      :+:    :+:            */
+/*   By: aholster <aholster@student.codam.nl>          +:+                    */
+/*       ivan-tey <ivan-tey@student.codam.nl>         +#+                     */
+/*       nloomans <nloomans@student.codam.nl>        +#+                      */
+/*                                                 #+#    #+#                 */
+/*   License: GPLv3                                ########   odam.nl         */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "private.h"
+#include <assert.h>
+
+int						env_set(t_env *const env,
+							char const *const key,
+							char const *const value)
+{
+	struct s_env_pair	*new;
+
+	assert(key != NULL || value != NULL);
+	env_unset(env, key);
+	new = ft_memalloc(sizeof(*new));
+	if (new == NULL)
+		return (-1);
+	new->key = ft_strdup(key);
+	if (new->key == NULL)
+	{
+		ft_memdel((void **)&new);
+		return (-1);
+	}
```

Comment:
error check can be merged into the next one.

Reply:
`	if (new->value == NULL || new->key == NULL)
	{
		ft_strdel(&new->key);
                ft_strdel(&new->value);
		ft_memdel((void **)&new);
		return (-1);
	}`